src/Executor/ReferenceExecutor.php

```diff
@@ -1092,6 +1092,7 @@ protected function completeAbstractValue(
         $contextValue
     ) {
         $typeCandidate = $returnType->resolveType($result, $contextValue, $info);
+        $result = $returnType->resolveValue($result, $contextValue, $info);
```

Comment:
What's going on with the pass-by-reference of `$result`? What happens if we remove `&`?

Also, what is going on with its type - we seem to know it is `array<mixed>` here, but is it really? What if me make it a hard type hint?

Reply:
This seems like a bigger issue to investigate, ideally, separately from this PR. 

If we ship this as is, I'll make sure to follow up on it.

Reply:
the existing resolveType function also hints for `mixed`: https://github.com/ruudk/graphql-php/blob/a753f1c56df97dfe105c1c2ac8abc6d72af2af28/src/Type/Definition/AbstractType.php#L14-L24